jas/runner/jasrunner.go

```diff
@@ -169,15 +170,15 @@ func runSastScan(securityParallelRunner *utils.SecurityParallelRunner, scanner *
 		defer func() {
 			securityParallelRunner.JasScannersWg.Done()
 		}()
-		results, err := sast.RunSastScan(scanner, module, signedDescriptions, threadId)
-		if err != nil {
-			return fmt.Errorf("%s %s", clientutils.GetLogMsgPrefix(threadId, false), err.Error())
-		}
+		vulnerabilitiesResults, violationsResults, err := sast.RunSastScan(scanner, module, signedDescriptions, threadId)
 		securityParallelRunner.ResultsMu.Lock()
 		defer securityParallelRunner.ResultsMu.Unlock()
-		extendedScanResults.SastScanResults = append(extendedScanResults.SastScanResults, results...)
-		err = dumpSarifRunToFileIfNeeded(results, scansOutputDir, jasutils.Sast)
-		return
+		// We first add the scan results and than check for errors to store the exit code to report it in the end
```

Comment:
```suggestion
		// We first add the scan results and only then check for errors, so we can store the exit code in order to report it in the end
```
Fixed typo and rephrased to better understand